src/operators/pm.cc

```diff
@@ -15,20 +15,87 @@
 
 #include "src/operators/pm.h"
 
-#include <string.h>
-
-#include <string>
 #include <algorithm>
 #include <iterator>
 #include <sstream>
-#include <vector>
-#include <list>
-#include <memory>
 
-#include "src/operators/operator.h"
 #include "src/utils/acmp.h"
 #include "src/utils/string.h"
 
+
+static inline std::string parse_pm_content(const std::string &op_parm) {
+    auto offset = op_parm.find_first_not_of(" \t");
+    if (offset == std::string::npos) {
+        return op_parm;
+    }
+
+    auto size = op_parm.size() - offset;
+    if (size >= 2 &&
+        op_parm.at(offset) == '\"' && op_parm.back() == '\"') {
+        offset++;
+        size -= 2;
+    }
+
+    if (size == 0) {
+        return op_parm;
+    }
+
+    std::string parsed_parm(op_parm.c_str() + offset, size);
+
+    unsigned char bin_offset = 0;
+    unsigned char bin_parm[3] = { 0 };
+    bool bin = false, esc = false;
+
+    char *d = parsed_parm.data();
+    for(const char *s = d, *e = d + size; s != e; ++s) {
+        if (*s == '|') {
+            bin = !bin;
+        } else if(!esc && *s == '\\') {
+            esc = true;
+        } else {
+            if (bin) {
+                if (*s == '0' || *s == '1' || *s == '2' ||
```

Comment:
Wouldn't it be simpler to use `isdigit()` like in [v2](https://github.com/owasp-modsecurity/ModSecurity/blob/v2/master/apache2/msc_util.c#L460)?

Reply:
Yes, you're right. I was mostly focused on removing the extra allocations in `parse_pm_content` and moving that function from `src/utils/acmp.cc` (where it was originally implemented), so I didn't focus on changing much else.

I'll replace all these comparisons with `VALID_HEX` (available in `src\utils\string.h`).

Reply:
I just pushed an update to the commit to use `VALID_HEX` and simplify this code.